### PR TITLE
fix(isolated_declarations): inferring literal types for readonly class fileds

### PR DESCRIPTION
**Pull request by escaton**

### CODE REVIEW

**Review by Dunqing — CHANGES_REQUESTED** [review]

File: crates/oxc_isolated_declarations/src/class.rs [hunk 1]
```diff
@@ -76,6 +80,9 @@ impl<'a> IsolatedDeclarations<'a> {
                 })
         };
 
+        // TODO if inferred type_annotations is TSLiteral, it should stand as value,
+        // so `field = 'string'` remain `field = 'string'` instead of `field: 'string'`
```
[the remark this answers is not in the document]
Comment (escaton) [reply crates/oxc_isolated_declarations/src/class.rs:84]:
```
export class Baz {
  readonly prop = 'some string';
}
```
turns into this
```
export declare class Baz {
	readonly prop: 'some string';
}
```
Which is valid, but TSC emits
```
export declare class Baz {
    readonly prop = "some string";
}
```

[the remark this answers is not in the document]
Comment (Dunqing) [reply crates/oxc_isolated_declarations/src/class.rs:84]: Thank you for explaining this. I get it.

[the remark this answers is not in the document]
Comment (Dunqing) [reply crates/oxc_isolated_declarations/src/class.rs:84]: If you want to fix this then you can refer
https://github.com/oxc-project/oxc/blob/78c66d3f059a6c964c9e85a58314eb905a3723fe/crates/oxc_isolated_declarations/src/declaration.rs#L73-L79

**Review by Dunqing — APPROVED** [review]

**Comment by Dunqing** [comment]
Thank you!

**Comment by escaton** [comment]
@Dunqing how can I solve the problem with MQ?

**Pull request merged**